docs/installation/webserver/nginx.md

```diff
@@ -6,106 +6,263 @@ Nginx: Configuring the virtual host
-    }
+    server_name                example.com;
+
+    # Logging
+    access_log                 /var/log/nginx/example.com.access.log;
+    error_log                  /var/log/nginx/example.com.error.log;
+
+    # Site root
+    root                       /var/www/sites/example.com/public;
+    index                      index.php;
+
+    # Bolt specific
+    include                    global/bolt.conf;
+
+    # PHP FPM
+    include                    global/fpm.conf;
+
+    # Let's Encrypt Certbot challenges (optional)
+    #include                    global/acme-challenge.conf;
+
+    # Restrictions
+    include                    global/restrictions.conf;
 }
 ```
 
-**Note:** 502 Bad Gateway Errors
 
-If you're using UNIX sockets instead of TCP ports on your PHP-FPM installation,
-you will need to change the `fastcgi_pass` parameters to match what is set in
-your PHP-FPM configuration's `listen = `, e.g.:
+**NOTE:** You will need to customise/change the values for the `server_name`,
+`access_log`, `error_log` and `root` parameters to match the domain name and
+path locations relevant to your system.
+
+
+### Common
+
+Common, or "global", configuration settings are stored in a directory under
+`/etc/nginx/`. For our examples we use the `/etc/nginx/global/` directory for
+no other reason besides semantics.
+
+This section contains details on the following files:
+
+| File name             | Description |
+| --------------------- | ----------- |
+| `bolt.conf`           | Bolt specific routes
+| `restrictions.conf`   | Files & directories to block access to
+| `php-fpm.conf`        | PHP-FPM configuration
+| `acme-challenge.conf` | (optional) Let's Encrypt Certbot challenges
+
+
+#### `bolt.conf`
 
+The `bolt.conf` file define location maches common to all of your Bolt sites on
+a host.
+
+```nginx
+# Redirect requests for */index.php to the same route minus the "index.php"
+if ($request_uri ~* "^(.*/)index\.php(?>/|)(.*)$") {
+    return                     301 $1$2;
+}
+
+# Default prefix match, as all URIs begin with /
+location / {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Bolt backend access
+#
+# NOTE: If you set a custom branding path, you will need to change '/bolt/'
+#       here to match
+location ~* /bolt/(.*)$ {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Generated thumbnail images
+location ~* /thumbs/(.*)$ {
+    access_log                 off;
+    log_not_found              off;
+    expires                    max;
+    add_header                 Pragma public;
+    add_header                 Cache-Control "public, mustrevalidate, proxy-revalidate";
+
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Don't log, and do cache, asset files
+location ~* ^.+\.(?:atom|bmp|bz2|css|doc|eot|exe|gif|gz|ico|jpe?g|jpeg|jpg|js|map|mid|midi|mp4|ogg|ogv|otf|png|ppt|rar|rtf|svg|svgz|tar|tgz|ttf|wav|woff|woff2|xls|zip)$ {
```

Comment:
woff2 is not in nginx default mime.types file, so we probably shouldn't configure it here either (or add a note saying to add it to mime.types)